renderer/modules/shared/components/drawer/navigation/custom-drawer/custom-drawer.tsx

```diff
@@ -1,17 +1,103 @@
+import { useState, useEffect } from 'react';
 import { useRouter } from 'next/router';
 import { drawerLinks, DrawerLink } from './drawer-links';
 import List from '@material-ui/core/List';
 import SectionLink from '../section-link/section-link';
 import useDawerStyles from '../../drawer.style';
-import UpgradeBanner from '../../../upgrade-banner/upgrade-banner';
+import UpgradeBanner, {
+  UpgradeBannerProps,
+} from '../../../upgrade-banner/upgrade-banner';
 import ProfilePicture from '../../../profile-picture/profile-picture';
+import { IpcRendererEvent } from 'electron';
 
 interface CustomDrawerProps {
   classes: ReturnType<typeof useDawerStyles>;
 }
 
+interface IdeDistribution {
+  id: string;
+  ideConfig: {
+    basepath: string;
+    workspaces: string;
+    commands: string;
+    version: string;
+  };
+}
+
+interface DevonIdeScripts {
+  version: string;
+  updated: Date;
+}
 export default function CustomDrawer(props: CustomDrawerProps): JSX.Element {
   const router = useRouter();
+  const initialProps: UpgradeBannerProps = {
+    version: '',
+    infoText: '',
+  };
+  const [displayUpgradeBanner, setDisplayUpgradeBanner] = useState(false);
+  const [upgradeBannerProps, setUpgradeBannerProps] = useState<
+    UpgradeBannerProps
+  >(initialProps);
+
+  const getLocalLatestVersion = (versions: string[]): string => {
+    const numericVersions = versions.map((v) => {
+      return +v.split('.').join(''); // for input "2020.04.001" it will return 202004001
+    });
+    const maxVersion = Math.max(...numericVersions); // for input [202008001, 202004002, 202004004] it will return 202008001
+    const latestLocalVersion = versions.filter(
+      (v) => +v.split('.').join('') === maxVersion
+    );
+    return latestLocalVersion[0];
+  };
+
+  const updateRequired = (
+    localVersions: string[],
+    latestVersion: string
+  ): void => {
+    if (!localVersions.includes(latestVersion)) {
+      const props: UpgradeBannerProps = {
+        version: latestVersion,
+        infoText:
+          'Latest version installed in your system is ' +
+          getLocalLatestVersion(localVersions),
+      };
+      setDisplayUpgradeBanner(true);
+      setUpgradeBannerProps({ ...props });
+    }
+  };
+
+  const getLocalInstalledVersions = (mavenLatestVersion: string): void => {
+    global.ipcRenderer.send('find:devonfwInstances');
+    global.ipcRenderer.on(
+      'get:devoninstances',
+      (_: IpcRendererEvent, data: IdeDistribution[]) => {
+        if (data) {
+          const localVersions = data.map((i) => i.ideConfig.version);
+          updateRequired(localVersions, mavenLatestVersion);
+        }
+      }
+    );
+  };
+
+  const getLatesDevonIdeVersion = (): void => {
+    global.ipcRenderer.send('find:latestDevonIdeScript');
+    global.ipcRenderer.on(
+      'get:latestDevonIdeScript',
+      (_: IpcRendererEvent, data: DevonIdeScripts) => {
+        if (data) getLocalInstalledVersions(data.version);
+      }
+    );
+  };
```

Comment:
This logic should be handled in the "backend". Here you could call to *find:check-for-updates*, and then receive something like 
```ts
{ updateAvailable: true, latestLocalVersion:12345, latestVersion: 1235}
```

Reply:
This way we avoid bloating the application with listeners and, in general, in a frontend application you need to display a complex state, you should receive it directly instead of composing it yourself

Here you are
1. Asking fot the local latest version
2. Asking fot the latest devonfw ide version
3. Formatting the versions (split, join)
4. Checking with one is greater
5. Manage multiple unsubscriptions

If you receive updateAvailable, latestLocalVersion and latestVersion, you have to;
1. Check update
2. Manage one unsubscription.

I know that the other steps would be done in the backend, but thats exactly his task: Processing the data so you dont have to do it in the frontend